src/beamlime/core/protocols.py

```diff
@@ -1,6 +1,7 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
 
+from asyncio import Task
 from typing import Any, Protocol, runtime_checkable
 
 
```

Comment:
I cannot comment on L9, but does it make sense to rename `BeamlimeApplicationControlProtocol` to `BeamlimeControlProtocol`? I got confused while reading between `BeamlimeApplicationControlProtocol` and `BeamlimeApplicationProtocol` (I just saw something starting with `BeamlimeApplication` and ending in `Protocol` and started mixing them up).

No worries if you think it's necessary to keep it as is.